posthog/hogql_queries/insights/funnels/base.py

```diff
@@ -315,6 +315,25 @@ def _serialize_step(
     def extra_event_fields_and_properties(self):
         return self._extra_event_fields + self._extra_event_properties
 
+    @property
+    def _absolute_actors_step(self) -> Optional[int]:
+        """The actor query's 1-indexed target step converted to our 0-indexed SQL form. Never a negative integer."""
+        if self.context.actorsQuery is None:
+            return None
+        target_step = self.context.actorsQuery.funnelStep
+        if target_step is None:
+            return None
+        if target_step < 0:
+            # the first valid dropoff argument for funnel_step is -2
+            # -2 refers to persons who performed the first step but never made it to the second
+            if target_step == -1:
+                raise ValueError("To request dropoff of initial step use -2")
```

Comment:
```suggestion
            if target_step == -1:
                raise ValueError("The first valid drop-off argument for funnelStep is -2. -2 refers to persons who performed the first step but never made it to the second.")
```
I'd honestly just put the comment into the error message, as it's easier to understand for someone without context.